lib/dependencies.sh

```diff
@@ -307,3 +316,28 @@ npm_prune_devdependencies() {
     meta_set "skipped-prune" "false"
   fi
 }
+
+pnpm_prune_devdependencies() {
+  local pnpm_version
+  local build_dir=${1:-}
+
+  pnpm_version=$(pnpm --version)
+
+  if [ "$NODE_ENV" == "test" ]; then
+    echo "Skipping because NODE_ENV is 'test'"
+    meta_set "skipped-prune" "true"
+    return 0
+  elif [ "$NODE_ENV" != "production" ]; then
+    echo "Skipping because NODE_ENV is not 'production'"
+    meta_set "skipped-prune" "true"
+    return 0
+  elif [ -n "$NPM_CONFIG_PRODUCTION" ]; then
```

Comment:
I know you've copied this from above, but it doesn't make sense to me! It's saying that if the `NPM_CONFIG_PRODUCTION` env var isn't empty then we won't prune dev dependencies. But the [docs](https://devcenter.heroku.com/articles/nodejs-support#skip-pruning) say that it will only skip pruning if `NPM_CONFIG_PRODUCTION=false`!

Reply:
Good point, I applied the `YARN2` approach and defined a `PNPM_SKIP_PRUNING` variable.